src/components/Vega.svelte

```diff
@@ -166,6 +167,39 @@
         Object.entries(signals).forEach(([key, v]) => {
           spec.signals.push({ name: key, value: v });
         });
+        const signalNames = spec.signals.map((signal) => signal.name);
+        if (!signalNames['width']) {
```

Comment:
isn't signal names an array. Thus, shouldn't it be `!signalNames.includes('width')`?

Reply:
Good catch. It turns out the checking was extraneous anyway, so it always passed.  I'll remove it.  

I wasn't clear on how often the updateSpec might be called, and was confused by another duplicate signal error.